luigi/scheduler.py

```diff
@@ -804,6 +889,20 @@ def get_work(self, host=None, assistant=False, current_tasks=None, worker=None,
             if self._schedulable(task) and self._has_resources(task.resources, greedy_resources):
                 if in_workers and self._has_resources(task.resources, used_resources):
                     best_task = task
+                    batch_param_names, max_batch_size = self._state.get_batcher(
+                        worker_id, task.family)
+                    if batch_param_names and task.is_batchable():
+                        try:
+                            batched_params = {
+                                name: [task.params[name]] for name in batch_param_names
```

Comment:
It's a dictionary comprehension of singleton lists. It has the list of batch parameters for each parameter name. When we first create it, each batch parameter only has one value.